web/common/utils/timeSeriesUtils.ts

```diff
@@ -44,45 +35,39 @@ const mapStatForDisplay = (stat: string): string => {
   }
 };
 
-const nameHistogramStat = (matrix: TimeSeries[], stat: string, labelPrettifier?: KVMapper): TimeSeries[] => {
-  const statDisplay = mapStatForDisplay(stat);
-  matrix.forEach(ts => {
-    const labels = Object.keys(ts.labelSet)
-      .filter(k => k !== 'reporter')
-      .map(k => {
-        const val = ts.labelSet[k];
-        return labelPrettifier ? labelPrettifier(k, val) : val;
-      })
-      .join(',');
-    if (labels === '') {
-      // Ex: average // quantile 0.999 // etc.
-      ts.name = statDisplay;
-    } else {
-      // Ex: policy: average // stadium: quantile 0.999 // etc.
-      ts.name = `${labels}: ${statDisplay}`;
-    }
-  });
-  return matrix;
-};
-
-const nameTimeSeries = (matrix: TimeSeries[], chartName: string, labelPrettifier?: KVMapper): TimeSeries[] => {
-  matrix.forEach(ts => {
-    const labels = Object.keys(ts.labelSet)
-      .filter(k => k !== 'reporter')
+const nameTimeSeries = (series: TimeSeries[], labelPrettifier?: KVMapper): NamedTimeSeries[] => {
+  let hasSeveralFamilyNames = false;
+  if (series.length > 0) {
+    const firstName = series[0].labelSet[nameLabel];
+    hasSeveralFamilyNames = series.some(s => s.labelSet[nameLabel] !== firstName);
+  }
+  return series.map(ts => {
+    const name = ts.labelSet[nameLabel];
+    const stat = mapStatForDisplay(ts.labelSet[statLabel]);
+    const otherLabels = Object.keys(ts.labelSet)
+      .filter(k => k !== nameLabel && k !== statLabel)
```

Comment:
Filtering out the `reporter` label is removed. Is it no longer necessary?
In a quick scan to the go-changes is unclear for me.

Reply:
perhaps a mistake, let me check in kiali ...

Reply:
Actually, this was an old filter that has turned useless a while ago, it can be safely removed. At that time, we used to fetch metrics for both reporters (grouped by reporter) that why it was appearing in labels and had to be filtered out. Today, metrics are fetched *filtered* by reporter, so that it doesn't appear in labels.

And anyway, even if we had to do something about reporters, it should be in kiali, not in k-charted, because k-charted doesn't assume metrics are the istio ones.